File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/basesection/fragments/LeftMenu.java
package com.baytrendy.shopifyapp.basesection.fragments;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000R\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\u0010\u000e\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0006\u0018\u0000 !2\u00020\u0001:\u0002 !B\u0005\u00a2\u0006\u0002\u0010\u0002J\u001c\u0010\r\u001a\u00020\u000e2\u0012\u0010\u000f\u001a\u000e\u0012\u0004\u0012\u00020\u0011\u0012\u0004\u0012\u00020\u00110\u0010H\u0002J\u0010\u0010\u0012\u001a\u00020\u000e2\u0006\u0010\u0013\u001a\u00020\u0014H\u0016J&\u0010\u0015\u001a\u0004\u0018\u00010\u00162\u0006\u0010\u0017\u001a\u00020\u00182\b\u0010\u0019\u001a\u0004\u0018\u00010\u001a2\b\u0010\u001b\u001a\u0004\u0018\u00010\u001cH\u0016J\b\u0010\u001d\u001a\u00020\u000eH\u0016J\b\u0010\u001e\u001a\u00020\u000eH\u0016J\b\u0010\u001f\u001a\u00020\u000eH\u0016R\u0010\u0010\u0003\u001a\u0004\u0018\u00010\u0004X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0005\u001a\u0004\u0018\u00010\u0006X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u001e\u0010\u0007\u001a\u00020\b8\u0006@\u0006X\u0087.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\t\u0010\n\"\u0004\b\u000b\u0010\f\u00a8\u0006\""}, d2 = {"Lcom/shopify/shopifyapp/basesection/fragments/LeftMenu;", "Lcom/shopify/shopifyapp/basesection/fragments/BaseFragment;", "()V", "binding", "Lcom/shopify/shopifyapp/base/databinding/MLeftmenufragmentBinding;", "currentactivity", "Landroid/app/Activity;", "viewModelFactory", "Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "getViewModelFactory", "()Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "setViewModelFactory", "(Lcom/shopify/shopifyapp/utils/ViewModelFactory;)V", "consumeResponse", "", "hash", "Ljava/util/HashMap;", "", "onAttach", "context", "Landroid/content/Context;", "onCreateView", "Landroid/view/View;", "inflater", "Landroid/view/LayoutInflater;", "container", "Landroid/view/ViewGroup;", "savedInstanceState", "Landroid/os/Bundle;", "onDestroyView", "onDetach", "onResume", "ClickHandlers", "Companion", "base_debug"})
public final class LeftMenu extends com.baytrendy.shopifyapp.basesection.fragments.BaseFragment {
    private com.baytrendy.shopifyapp.databinding.MLeftmenufragmentBinding binding;
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Inject()
    public com.baytrendy.shopifyapp.utils.ViewModelFactory viewModelFactory;
    private android.app.Activity currentactivity;
    @org.jetbrains.annotations.NotNull()
    public static androidx.appcompat.widget.LinearLayoutCompat menulist;
    private static android.content.Context currentcontext;
    @org.jetbrains.annotations.NotNull()
    protected static com.baytrendy.shopifyapp.basesection.viewmodels.LeftMenuViewModel leftmenu;
    private static com.baytrendy.shopifyapp.basesection.models.MenuData menuData;
    public static final com.baytrendy.shopifyapp.basesection.fragments.LeftMenu.Companion Companion = null;
    private java.util.HashMap _$_findViewCache;
    
    @org.jetbrains.annotations.NotNull()
    public final com.baytrendy.shopifyapp.utils.ViewModelFactory getViewModelFactory() {
        return null;
    }
    
    public final void setViewModelFactory(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.utils.ViewModelFactory p0) {
    }
    
    @org.jetbrains.annotations.Nullable()
    @java.lang.Override()
    public android.view.View onCreateView(@org.jetbrains.annotations.NotNull()
    android.view.LayoutInflater inflater, @org.jetbrains.annotations.Nullable()
    android.view.ViewGroup container, @org.jetbrains.annotations.Nullable()
    android.os.Bundle savedInstanceState) {
        return null;
    }
    
    private final void consumeResponse(java.util.HashMap<java.lang.String, java.lang.String> hash) {
    }
    
    @java.lang.Override()
    public void onAttach(@org.jetbrains.annotations.NotNull()
    android.content.Context context) {
    }
    
    @java.lang.Override()
    public void onDetach() {
    }
    
    @java.lang.Override()
    public void onDestroyView() {
    }
    
    @java.lang.Override()
    public void onResume() {
    }
    
    public LeftMenu() {
        super();
    }
    
    @kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000.\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0010\u000b\n\u0000\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\u0018\u00002\u00020\u0001B\u000f\u0012\b\u0010\u0002\u001a\u0004\u0018\u00010\u0003\u00a2\u0006\u0002\u0010\u0004J\u000e\u0010\n\u001a\u00020\u000b2\u0006\u0010\f\u001a\u00020\rJ\u0016\u0010\u000e\u001a\u00020\u000b2\u0006\u0010\f\u001a\u00020\r2\u0006\u0010\u000f\u001a\u00020\u0010J\u000e\u0010\u0011\u001a\u00020\u000b2\u0006\u0010\f\u001a\u00020\rR\u001c\u0010\u0002\u001a\u0004\u0018\u00010\u0003X\u0080\u000e\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u0005\u0010\u0006\"\u0004\b\u0007\u0010\u0004R\u000e\u0010\b\u001a\u00020\tX\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u0012"}, d2 = {"Lcom/shopify/shopifyapp/basesection/fragments/LeftMenu$ClickHandlers;", "", "context", "Landroid/content/Context;", "(Landroid/content/Context;)V", "getContext$base_debug", "()Landroid/content/Context;", "setContext$base_debug", "open", "", "expandMenu", "", "view", "Landroid/view/View;", "getMenu", "menudata", "Lcom/shopify/shopifyapp/basesection/models/MenuData;", "navigationClicks", "base_debug"})
    public static final class ClickHandlers {
        private boolean open;
        @org.jetbrains.annotations.Nullable()
        private android.content.Context context;
        
        public final void getMenu(@org.jetbrains.annotations.NotNull()
        android.view.View view, @org.jetbrains.annotations.NotNull()
        com.baytrendy.shopifyapp.basesection.models.MenuData menudata) {
        }
        
        public final void expandMenu(@org.jetbrains.annotations.NotNull()
        android.view.View view) {
        }
        
        public final void navigationClicks(@org.jetbrains.annotations.NotNull()
        android.view.View view) {
        }
        
        @org.jetbrains.annotations.Nullable()
        public final android.content.Context getContext$base_debug() {
            return null;
        }
        
        public final void setContext$base_debug(@org.jetbrains.annotations.Nullable()
        android.content.Context p0) {
        }
        
        public ClickHandlers(@org.jetbrains.annotations.Nullable()
        android.content.Context context) {
            super();
        }
    }
    
    @kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000<\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\b\u0086\u0003\u0018\u00002\u00020\u0001B\u0007\b\u0002\u00a2\u0006\u0002\u0010\u0002J\u000e\u0010\u0013\u001a\u00020\u00142\u0006\u0010\u0015\u001a\u00020\u0016J\u0018\u0010\u0017\u001a\u00020\u00142\u0006\u0010\u0018\u001a\u00020\u00192\u0006\u0010\r\u001a\u00020\u000eH\u0002R\u0010\u0010\u0003\u001a\u0004\u0018\u00010\u0004X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u001a\u0010\u0005\u001a\u00020\u0006X\u0084.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u0007\u0010\b\"\u0004\b\t\u0010\nR\u0010\u0010\u000b\u001a\u0004\u0018\u00010\fX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u001a\u0010\r\u001a\u00020\u000eX\u0086.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u000f\u0010\u0010\"\u0004\b\u0011\u0010\u0012\u00a8\u0006\u001a"}, d2 = {"Lcom/shopify/shopifyapp/basesection/fragments/LeftMenu$Companion;", "", "()V", "currentcontext", "Landroid/content/Context;", "leftmenu", "Lcom/shopify/shopifyapp/basesection/viewmodels/LeftMenuViewModel;", "getLeftmenu", "()Lcom/shopify/shopifyapp/basesection/viewmodels/LeftMenuViewModel;", "setLeftmenu", "(Lcom/shopify/shopifyapp/basesection/viewmodels/LeftMenuViewModel;)V", "menuData", "Lcom/shopify/shopifyapp/basesection/models/MenuData;", "menulist", "Landroidx/appcompat/widget/LinearLayoutCompat;", "getMenulist", "()Landroidx/appcompat/widget/LinearLayoutCompat;", "setMenulist", "(Landroidx/appcompat/widget/LinearLayoutCompat;)V", "renderSuccessResponse", "", "data", "Lcom/google/gson/JsonElement;", "updateMenu", "array", "Lorg/json/JSONArray;", "base_debug"})
    public static final class Companion {
        
        @org.jetbrains.annotations.NotNull()
        public final androidx.appcompat.widget.LinearLayoutCompat getMenulist() {
            return null;
        }
        
        public final void setMenulist(@org.jetbrains.annotations.NotNull()
        androidx.appcompat.widget.LinearLayoutCompat p0) {
        }
        
        @org.jetbrains.annotations.NotNull()
        protected final com.baytrendy.shopifyapp.basesection.viewmodels.LeftMenuViewModel getLeftmenu() {
            return null;
        }
        
        protected final void setLeftmenu(@org.jetbrains.annotations.NotNull()
        com.baytrendy.shopifyapp.basesection.viewmodels.LeftMenuViewModel p0) {
        }
        
        public final void renderSuccessResponse(@org.jetbrains.annotations.NotNull()
        com.google.gson.JsonElement data) {
        }
        
        private final void updateMenu(org.json.JSONArray array, androidx.appcompat.widget.LinearLayoutCompat menulist) {
        }
        
        private Companion() {
            super();
        }
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/basesection/fragments/LeftMenu.kt
package com.baytrendy.shopifyapp.basesection.fragments

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.os.Bundle
import android.os.Handler
import android.util.Base64
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.appcompat.widget.LinearLayoutCompat
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import com.google.gson.JsonElement
import com.google.zxing.integration.android.IntentIntegrator
import com.mindorks.paracamera.Camera
import com.baytrendy.shopifyapp.MyApplication
import com.baytrendy.shopifyapp.addresssection.activities.AddressList
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.databinding.MDynamicmenuBinding
import com.baytrendy.shopifyapp.databinding.MLeftmenufragmentBinding
import com.baytrendy.shopifyapp.basesection.activities.NewBaseActivity
import com.baytrendy.shopifyapp.basesection.activities.Weblink
import com.baytrendy.shopifyapp.basesection.models.FeaturesModel
import com.baytrendy.shopifyapp.basesection.models.MenuData
import com.baytrendy.shopifyapp.basesection.viewmodels.LeftMenuViewModel
import com.baytrendy.shopifyapp.basesection.viewmodels.SplashViewModel.Companion.featuresModel
import com.baytrendy.shopifyapp.cartsection.activities.CartList
import com.baytrendy.shopifyapp.collectionsection.activities.CollectionList
import com.baytrendy.shopifyapp.collectionsection.activities.CollectionListMenu
import com.baytrendy.shopifyapp.livepreviewsection.LivePreview
import com.baytrendy.shopifyapp.loginsection.activity.LoginActivity
import com.baytrendy.shopifyapp.ordersection.activities.OrderList
import com.baytrendy.shopifyapp.productsection.activities.ProductList
import com.baytrendy.shopifyapp.productsection.activities.ProductView
import com.baytrendy.shopifyapp.searchsection.activities.AutoSearch
import com.baytrendy.shopifyapp.searchsection.activities.SearchByScanner
import com.baytrendy.shopifyapp.userprofilesection.activities.UserProfile
import com.baytrendy.shopifyapp.utils.Constant
import com.baytrendy.shopifyapp.utils.ViewModelFactory
import com.baytrendy.shopifyapp.wishlistsection.activities.WishList
import org.json.JSONArray
import org.json.JSONObject
import java.nio.charset.Charset
import java.util.HashMap
import javax.inject.Inject

class LeftMenu : BaseFragment() {
    private var binding: MLeftmenufragmentBinding? = null

    @Inject
    lateinit var viewModelFactory: ViewModelFactory

    private var currentactivity: Activity? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        binding = DataBindingUtil.inflate(context!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater, R.layout.m_leftmenufragment, container, true)
        menulist = binding!!.menulist
        var pInfo: PackageInfo? = null
        try {
            pInfo = activity!!.packageManager.getPackageInfo(activity!!.packageName, 0)
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }

        val version = pInfo!!.versionName
        val versioncode = pInfo.versionCode
        Log.i("MageNative", "LeftMenuResume 4")
        menuData = MenuData()
        val app_version = "App Version $version($versioncode)"
        menuData!!.appversion = app_version
        menuData!!.copyright = resources.getString(R.string.copy) + " " + resources.getString(R.string.app_name)
        binding!!.features = featuresModel
        binding!!.menudata = menuData
        binding!!.clickdata = ClickHandlers(currentcontext)
        (activity!!.application as MyApplication).mageNativeAppComponent!!.doLeftMeuInjection(this)
        leftmenu = ViewModelProvider(this, viewModelFactory).get(LeftMenuViewModel::class.java)
        leftmenu.data.observe(viewLifecycleOwner, Observer<HashMap<String, String>> { this.consumeResponse(it) })
        return binding!!.root
    }

    private fun consumeResponse(hash: HashMap<String, String>) {
        menuData!!.tag = hash.get("tag")
        Log.i("MageNative", "LeftMenuResume 3" + hash["firstname"]!!)
        menuData!!.username = hash["firstname"] + " " + hash["secondname"]
        if (hash["tag"] == "login") {
            menuData!!.visible = View.VISIBLE
        } else {
            menuData!!.visible = View.GONE
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        currentcontext = context
        currentactivity = context as NewBaseActivity
    }

    override fun onDetach() {
        super.onDetach()
    }

    override fun onDestroyView() {
        super.onDestroyView()
    }

    class ClickHandlers(internal var context: Context?) {
        private var open = false

        fun getMenu(view: View, menudata: MenuData) {
            when (menudata.type) {
                "collection" -> {
                    val intent = Intent(context, ProductList::class.java)
                    if (menudata.id == null) {
                        intent.putExtra("handle", menudata.handle)
                    } else {
                        try {
                            val data = Base64.encode(("gid://shopify/Collection/" + menudata.id!!).toByteArray(), Base64.DEFAULT)
                            val id = String(data, Charset.defaultCharset()).trim { it <= ' ' }
                            intent.putExtra("ID", id)
                        } catch (e: Exception) {
                            e.printStackTrace()
                        }

                    }
                    intent.putExtra("tittle", menudata.title)
                    context!!.startActivity(intent)
                    Constant.activityTransition(context!!)
                }
                "product" -> {
                    val productintent = Intent(context, ProductView::class.java)
                    if (menudata.id == null) {
                        productintent.putExtra("handle", menudata.handle)
                    } else {
                        try {
                            val data = Base64.encode(("gid://shopify/Product/" + menudata.id!!).toByteArray(), Base64.DEFAULT)
                            val id = String(data, Charset.defaultCharset()).trim { it <= ' ' }
                            productintent.putExtra("ID", id)
                        } catch (e: Exception) {
                            e.printStackTrace()
                        }

                    }
                    productintent.putExtra("tittle", menudata.title)
                    context!!.startActivity(productintent)
                    Constant.activityTransition(context!!)
                }
                "product-all" -> {
                    val product_all = Intent(context, ProductList::class.java)
                    product_all.putExtra("tittle", menudata.title)
                    context!!.startActivity(product_all)
                    Constant.activityTransition(context!!)
                }
                "collection-all" -> {
                    val collection_all = Intent(context, CollectionList::class.java)
                    context!!.startActivity(collection_all)
                    Constant.activityTransition(context!!)
                }
                "page" -> {
                    val page = Intent(context, Weblink::class.java)
                    page.putExtra("name", menudata.title)
                    page.putExtra("link", menudata.url)
                    context!!.startActivity(page)
                    Constant.activityTransition(context!!)
                }
                "blog" -> {
                    val blog = Intent(context, Weblink::class.java)
                    blog.putExtra("name", menudata.title)
                    blog.putExtra("link", menudata.url)
                    context!!.startActivity(blog)
                    Constant.activityTransition(context!!)
                }
            }
        }

        fun expandMenu(view: View) {
            val constraintLayout = view.parent as ConstraintLayout
            val linearLayoutCompat = constraintLayout.getChildAt(2) as LinearLayoutCompat
            if (open) {
                linearLayoutCompat.visibility = View.GONE
                open = false
            } else {
                linearLayoutCompat.visibility = View.VISIBLE
                linearLayoutCompat.requestFocus()
                open = true
            }
        }

        fun navigationClicks(view: View) {
            when (view.tag as String) {
                "livepreview" -> {
                    val integrator = IntentIntegrator((context as NewBaseActivity)!!)
                    integrator.setPrompt("Scan Your Barcode")
                    integrator.setCameraId(0) // Use a specific camera of the device
                    integrator.setOrientationLocked(true)
                    integrator.setBeepEnabled(true)
                    integrator.captureActivity = LivePreview::class.java
                    integrator.initiateScan()
                }
                "currencyswitcher" -> {
                    Log.i("MageNative", "currencyswitcher" + " : IN")
                    (context as NewBaseActivity)!!.getCurrency()

                }
                "collections" -> {
                    val collection_all = Intent(context, CollectionList::class.java)
                    context!!.startActivity(collection_all)
                    Constant.activityTransition(context!!)
                }
                "Sign In" -> {
                    val login = Intent(context, LoginActivity::class.java)
                    context!!.startActivity(login)
                    Constant.activityTransition(context!!)
                }
                "mywishlist" -> {
                    val mywishlist = Intent(context, WishList::class.java)
                    context!!.startActivity(mywishlist)
                    Constant.activityTransition(context!!)
                }
                "mycartlist" -> {
                    val mycartlist = Intent(context, CartList::class.java)
                    context!!.startActivity(mycartlist)
                    Constant.activityTransition(context!!)
                }
                "invitefriends" -> {
                    val appPackageName = view.context.packageName // getPackageName() from Context or Activity object
                    val shareIntent = Intent(Intent.ACTION_SEND)
                    shareIntent.type = "text/plain"
                    shareIntent.putExtra(Intent.EXTRA_SUBJECT, view.context.resources.getString(R.string.app_name))
                    shareIntent.putExtra(Intent.EXTRA_TEXT, "https://play.google.com/store/apps/details?id=$appPackageName")
                    view.context.startActivity(Intent.createChooser(shareIntent, view.context.resources.getString(R.string.shareproduct)))
                    Constant.activityTransition(view.context)
                }
                "autosearch" -> {
                    val autosearch = Intent(context, AutoSearch::class.java)
                    context!!.startActivity(autosearch)
                    Constant.activityTransition(context!!)
                }
                "logout" -> {
                    Toast.makeText(context, context!!.resources.getString(R.string.successlogout), Toast.LENGTH_LONG).show()
                    leftmenu.logOut()
                }
                "myprofile" -> if (leftmenu.isLoggedIn) {
                    val myprofile = Intent(context, UserProfile::class.java)
                    context!!.startActivity(myprofile)
                    Constant.activityTransition(context!!)
                } else {
                    Toast.makeText(context, context!!.resources.getString(R.string.logginfirst), Toast.LENGTH_LONG).show()
                }
                "myorders" -> if (leftmenu.isLoggedIn) {
                    val myprofile = Intent(context, OrderList::class.java)
                    context!!.startActivity(myprofile)
                    Constant.activityTransition(context!!)
                } else {
                    Toast.makeText(context, context!!.resources.getString(R.string.logginfirst), Toast.LENGTH_LONG).show()
                }
                "myaddress" -> if (leftmenu.isLoggedIn) {
                    val myaddress = Intent(context, AddressList::class.java)
                    context!!.startActivity(myaddress)
                    Constant.activityTransition(context!!)
                } else {
                    Toast.makeText(context, context!!.resources.getString(R.string.logginfirst), Toast.LENGTH_LONG).show()
                }
            }
        }
    }

    override fun onResume() {
        super.onResume()
        Log.i("MageNative", "LeftMenuResume")
        when (activity!!.getPackageName()) {
            "com.shopify.shopifyapp" -> {
                menuData!!.previewvislible = View.VISIBLE
            }
            else -> {
                menuData!!.previewvislible = View.GONE
            }
        }
        leftmenu.context=currentcontext
        leftmenu.fetchUserData()
    }

    companion object {
        lateinit var menulist: LinearLayoutCompat
        private var currentcontext: Context? = null
        protected lateinit var leftmenu: LeftMenuViewModel
        private var menuData: MenuData? = null
        fun renderSuccessResponse(data: JsonElement) {
            Log.i("MageNative:", "MenuData$data")
            if (menulist.childCount == 0) {
                val handler = Handler()
                val runnable = Runnable {
                    try {
                        val `object` = JSONObject(data.toString())
                        if (`object`.getBoolean("success")) {
                            if (`object`.has("data")) {
                                val array = `object`.getJSONArray("data")
                                if (array.length() > 0) {
                                    for (i in 0 until array.length()) {
                                        handler.post {
                                            try {
                                                // Log.i("MageNative","CurrentContext :"+currentcontext)
                                                val binding: MDynamicmenuBinding = DataBindingUtil.inflate(currentcontext!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater, R.layout.m_dynamicmenu, null, false)
                                                val menuData = MenuData()
                                                if (array.getJSONObject(i).has("id")) {
                                                    menuData.id = array.getJSONObject(i).getString("id")
                                                }
                                                if (array.getJSONObject(i).has("handle")) {
                                                    menuData.handle = array.getJSONObject(i).getString("handle")
                                                }
                                                if (array.getJSONObject(i).has("type")) {
                                                    menuData.type = array.getJSONObject(i).getString("type")
                                                }
                                                menuData.title = array.getJSONObject(i).getString("title")
                                                if (array.getJSONObject(i).has("url")) {
                                                    menuData.url = array.getJSONObject(i).getString("url")
                                                }
                                                binding.menudata = menuData
                                                binding.clickdata = ClickHandlers(currentcontext)
                                                if (array.getJSONObject(i).has("menus")) {
                                                    binding.root.findViewById<View>(R.id.expand_collapse).visibility = View.VISIBLE
                                                    updateMenu(array.getJSONObject(i).getJSONArray("menus"), binding.root.findViewById(R.id.submenus))
                                                }
                                                menulist.addView(binding.root)
                                            } catch (e: Exception) {
                                                Log.i("MageNative", "Error" + e.message)
                                                Log.i("MageNative", "Error" + e.cause)
                                                e.printStackTrace()
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
                Thread(runnable).start()
            }
        }

        private fun updateMenu(array: JSONArray, menulist: LinearLayoutCompat) {
            val handler = Handler()
            val runnable = Runnable {
                if (array.length() > 0) {
                    for (i in 0 until array.length()) {
                        handler.post {
                            try {
                                try {
                                    val binding = DataBindingUtil.inflate<MDynamicmenuBinding>(currentcontext!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater, R.layout.m_dynamicmenu, null, false)
                                    val menuData = MenuData()
                                    if (array.getJSONObject(i).has("id")) {
                                        menuData.id = array.getJSONObject(i).getString("id")
                                    }
                                    if (array.getJSONObject(i).has("handle")) {
                                        menuData.handle = array.getJSONObject(i).getString("handle")
                                    }
                                    menuData.title = array.getJSONObject(i).getString("title")
                                    if (array.getJSONObject(i).has("type")) {
                                        menuData.type = array.getJSONObject(i).getString("type")
                                    }
                                    if (array.getJSONObject(i).has("url")) {
                                        menuData.url = array.getJSONObject(i).getString("url")
                                    }
                                    binding.menudata = menuData
                                    binding.clickdata = ClickHandlers(currentcontext)
                                    if (array.getJSONObject(i).has("menus")) {
                                        binding.root.findViewById<View>(R.id.expand_collapse).visibility = View.VISIBLE
                                        updateMenu(array.getJSONObject(i).getJSONArray("menus"), binding.root.findViewById(R.id.submenus))
                                    }
                                    menulist.addView(binding.root)
                                } catch (e: Exception) {
                                    e.printStackTrace()
                                }

                            } catch (e: Exception) {
                                e.printStackTrace()
                            }
                        }
                    }
                }
            }
            Thread(runnable).start()
        }
    }

}
